Source/drlg_l4.cpp

```diff
@@ -832,7 +832,7 @@ void __cdecl L4makeDmt()
 	int v6;        // ebx
 
 	v0 = 1;
-	v1 = dungeon;
+	v1 = (char (*)[40])dungeon;
 	do {

```

Comment:
So many typecasts added, some even looking quite ugly.  Like this one which was added yearlier, 14 lines below (currently line 849).  Where there are two typecasts used on the same side of the assignment already.  

And line 844 looks suspicious too: typecasting an l-value to a pointer to / array of `_BYTE`.  While both `v1` and `v2` are arrays of 40 elements of the same type (ptrs to an array of chars).  Maybe it's them who needs to be changed?

Reply:
PS. I hope there will be more stabs on this:)

Reply:
I would more or less disregard these since they are in functions that haven't yet been cleaned up and are just added to glue the clean code up with the still dirty parts. Should look much better once they are also cleaned up.

Reply:
Ah, I see it's a different function than those listed in the description. OK!

Reply:
Yeah you can ignore them for now, it's temporary added to garbage functions so they still compile. Functions that are actually cleaned don't have them.

Reply:
@sskras I think it's been a while since I posted an update to the progress, it's been pretty good lately:
![image](https://user-images.githubusercontent.com/204594/55673479-e0285380-58a8-11e9-9780-c45880f74041.png)

Reply:
Thanks, what do the color means?

Reply:
blue = devilution unique lines
red = diablo unique lines
yellow = total lines of difference